### PR TITLE
No action when injection feature is disabled

### PR DESCRIPTION
**Pull request by jprinet**
Add a flag to explicitly enable the feature

### CODE REVIEW

**Review by wolfs — APPROVED** [review]
LGTM!

File: README.adoc [hunk 1]
```diff
@@ -96,12 +96,15 @@ The plugin can be configured to inject Build Scans into any Gradle or Maven buil
 
 This feature works by installing scripts and Jars on each agent depending on the environment variable settings.
 
+The feature can be enabled by setting the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION` environment variable.
+
 If the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION` environment variable is present, injection is activated for Gradle builds.
-This causes an init script is installed in the user home of each connected agent.
-The init script is removed when the environment variable is removed, thus deactivating injection for Gradle builds.
+This causes an init script to be installed in the user home of each connected agent.
+The init script is removed when `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION` is set to `off`, thus deactivating injection for Gradle builds.
```
Comment (wolfs) [inline README.adoc:103]: This change (and the next) isn't necessary any more, right?

Comment (jprinet) [reply README.adoc]: well spotted, still relevant but the variable needs to be unset. I updated the doc accordingly

**Review by wolfs — APPROVED** [review]
LGTM!

**Review by alextu — COMMENTED** [review]
only had comments regarding the logging

File: src/main/java/hudson/plugins/gradle/injection/BuildScanInjectionListener.java [hunk 2]
```diff
@@ -18,20 +18,32 @@ public class BuildScanInjectionListener extends ComputerListener {
...
             EnvVars envGlobal = c.buildEnvironment(listener);
-            EnvVars envComputer = c.getEnvironment();
+            if(isInjectionEnabled(envGlobal)) {
+                try {
+                    EnvVars envComputer = c.getEnvironment();
 
-            inject(c, envGlobal, envComputer);
+                    inject(c, envGlobal, envComputer);
+                } catch (IOException | InterruptedException e) {
+                    LOGGER.info("Error processing scan injection - " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (jprinet) [reply src/main/java/hudson/plugins/gradle/injection/BuildScanInjectionListener.java:42]: to give a chance to filter those out by default as it appears to pop ups regularly under certain conditions (custom $HOME)

File: src/main/java/hudson/plugins/gradle/injection/BuildScanInjectionListener.java [hunk 3]
```diff
@@ -18,20 +18,32 @@ public class BuildScanInjectionListener extends ComputerListener {
...
+                }
+            }
         } catch (IOException | InterruptedException e) {
-            LOGGER.info("Error processing scan injection - " + e.getMessage());
+            // nothing can be done
```
[the remark this answers is not in the document]
Comment (jprinet) [reply src/main/java/hudson/plugins/gradle/injection/BuildScanInjectionListener.java:46]: We could add a log message if the feature is enabled, but my impression is that if an error pops when getting the build environment, then your whole instance is collapsing

**Review by alextu — APPROVED** [review]

**Pull request merged**